test/parser/statement/Function.test.js

```diff
@@ -181,6 +198,23 @@ describe("parser", () => {
     });
 
     describe("sub declarations", () => {
+        it('recovers when using `end function` instead of `end sub`', () => {
+            const { tokens } = brs.lexer.Lexer.scan(`
+                sub Main()
+                    print "Hello world"
+                end function
+                
+                sub DoSomething()
+                
+                end sub 
+            `);
+            let { statements, errors } = parser.parse(tokens);
+            expect(errors.length).toBe(1);
+            expect(statements).toBeDefined();
+            expect(statements).not.toBeNull();
+            expect({ errors, statements }).toMatchSnapshot();
```

Comment:
I don't know why I've never thought to package these up this way - I'd looked into using multiple snapshots (which ostensibly works?) but this is super nice :heart: